### ISSUE TITLE
Exception: In PG::ResultSet#read error

### ISSUE THREAD

**Issue by crimson-knight** [imdrasil/jennifer.cr#408]
I'm getting this error when I try and create a simple model. Maybe I just have tired eyes, but I'm not understanding if I messed up my mappings or not.

Current error message: 
```bash
INSERT INTO "web_pages"("url", "website_id", "created_at", "updated_at") VALUES ($1, $2, $3, $4)  RETURNING "id" | ["https://www.teamgantt.com/pricing", 10, 2022-05-05 00:26:18.985558000 UTC, 2022-05-05 00:26:18.985558000 UTC]
COMMIT
Exception: In PG::ResultSet#read the column id returned a Int64 but a (Int32 | Nil) was expected. (DB::ColumnTypeMismatchError)
```

My model:
```crystal
class WebPage < ApplicationRecord
  with_timestamps

  mapping(
    id: Primary64,
    url: String,
    website_id: Int64,
    created_at: Time?,
    updated_at: Time?
  )

  belongs_to :website, Website
end
```

Coding I'm trying to run:
```crystal
unknown_url = "https://www.teamgantt.com"
website = Website.where { c("url") == base_url }.first
new_webpage = WebPage.create({url: unknown_url, website_id: website.id})
```

The first error looks like it's expecting a Primary32 instead of a Primary 64, so here's the result by changing the mapping:

```bash
INSERT INTO "web_pages"("url", "website_id", "created_at", "updated_at") VALUES ($1, $2, $3, $4)  RETURNING "id" | ["https://www.teamgantt.com/pricing", 10, 2022-05-05 00:31:45.828833000 UTC, 2022-05-05 00:31:45.828833000 UTC]
COMMIT
Exception: In PG::ResultSet#read the column id returned a Int64 but a (Int32 | Nil) was expected. (DB::ColumnTypeMismatchError)
```

I'm confused about why the error message says the id column should be Int32 OR Nil, but I'll give it a try and I get this:
```crystal
class WebPage < ApplicationRecord
  with_timestamps

  mapping(
    id: { type: Int32?, primary: true },
    url: String,
    website_id: Int64,
    created_at: Time?,
    updated_at: Time?
  )

  belongs_to :website, Website
end
```

```bash
Exception: null value in column "id" of relation "web_pages" violates not-null constraint.
Original query was:
INSERT INTO "web_pages"("id", "url", "website_id", "created_at", "updated_at") VALUES ($1, $2, $3, $4, $5)  | [nil, "https://www.teamgantt.com/pricing", 10, 2022-05-05 00:35:17.687356000 UTC, 2022-05-05 00:35:17.687356000 UTC] (Jennifer::BadQuery)
```

This error makes sense to me, the primary key shouldn't be nil but I thought maybe my schema was messed up some how.

So why can't I get my model to map the `id` column properly as the primary key and use this `#create` method?
I'm using Crystal 1.4.1

**Comment by cyangle** [comment]
@crimson-knight I couldn't reproduce your error with postgresql. Maybe you need to update the shard to latest master branch.

Migration:
```crystal
class AddWebTables < Jennifer::Migration::Base
  def up
    create_table(:websites) do |t|
      t.string :url, {:null => false}
    end

    create_table(:web_pages) do |t|
      t.string :url, {:null => false}
      t.reference :website
      t.timestamps true
    end
  end

  def down
    drop_table :websites
    drop_table :web_pages
  end
end
```
Schema:
```
jennifer_test=# \d websites
                                    Table "public.websites"
 Column |          Type          | Collation | Nullable |               Default                
--------+------------------------+-----------+----------+--------------------------------------
 id     | bigint                 |           | not null | nextval('websites_id_seq'::regclass)
 url    | character varying(254) |           | not null | 
Indexes:
    "websites_pkey" PRIMARY KEY, btree (id)
Referenced by:
    TABLE "web_pages" CONSTRAINT "fk_cr_59102c17b7" FOREIGN KEY (website_id) REFERENCES websites(id) ON UPDATE RESTRICT ON DELETE RESTRICT

jennifer_test=# \d web_pages
                                        Table "public.web_pages"
   Column   |            Type             | Collation | Nullable |                Default                
------------+-----------------------------+-----------+----------+---------------------------------------
 id         | bigint                      |           | not null | nextval('web_pages_id_seq'::regclass)
 url        | character varying(254)      |           | not null | 
 website_id | bigint                      |           |          | 
 created_at | timestamp without time zone |           |          | 
 updated_at | timestamp without time zone |           |          | 
Indexes:
    "web_pages_pkey" PRIMARY KEY, btree (id)
Foreign-key constraints:
    "fk_cr_59102c17b7" FOREIGN KEY (website_id) REFERENCES websites(id) ON UPDATE RESTRICT ON DELETE RESTRICT

```

Code:
```crystal
require "../spec/spec_helper"

class WebPage < Jennifer::Model::Base
  with_timestamps

  mapping(
    id: Primary64,
    url: String,
    website_id: Int64,
    created_at: Time?,
    updated_at: Time?
  )

  belongs_to :website, Website
end

class Website < Jennifer::Model::Base
  mapping(
    id: Primary64,
    url: String,
  )

  has_many :web_pages, WebPage
end

it "has 1 webpage" do
  void_transaction do
    base_url = "https://en.wikipedia.org/"
    website = Website.create(url: base_url)
    unknown_url = "https://www.teamgantt.com"
    new_webpage = WebPage.create({url: unknown_url, website_id: website.id})
    website = Website.where { c("url") == base_url }.first.not_nil!
    pp website
    pp website.web_pages

    website.web_pages.first.not_nil!.url.should eq(unknown_url)
  end
end
```

**Comment by imdrasil** [comment]
Agree with cyangle - likely you have an issue with shard version or misaligned schema and mapping. If after migration to the latest version you still experience this issue please provide you table schema and stack trace for the 1st exception on your description

**Comment by crimson-knight** [comment]
Thanks, both of you! I forgot to comment sooner, but I realized the issue was because I was using a schema setup by Rails which sets the foreign_key id to be an Int32 when it should have been an Int64 (which is what I had defined).

I'm putting together some notes on how to run Jennifer side by side with ActiveRecord so other devs have info like this on how to migrate from Rails/AR to Crystal using Jennifer.
